sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -252,6 +252,165 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
     }
   }
 
+  test("Create partitioned data source table with partitioning columns but no schema") {
+    import testImplicits._
+
+    withTempPath { dir =>
+      val pathToPartitionedTable = new File(dir, "partitioned")
+      val df = sparkContext.parallelize(1 to 10).map(i => (i, i.toString)).toDF("num", "str")
+      df.write.format("parquet").partitionBy("num").save(pathToPartitionedTable.getCanonicalPath)
+      val tabName = "tab1"
+      withTable(tabName) {
+        spark.sql(
+          s"""
+             |CREATE TABLE $tabName
+             |USING parquet
+             |OPTIONS (
+             |  path '$pathToPartitionedTable'
+             |)
+             |PARTITIONED BY (inexistentColumns)
+           """.stripMargin)
+        val tableMetadata = spark.sessionState.catalog.getTableMetadata(TableIdentifier(tabName))
```

Comment:
we can abstract common logic into some methods, to remove duplicated code a bit.